public/app/plugins/datasource/cloudwatch/__mocks__/CloudWatchDataSource.ts

```diff
@@ -1,62 +1,93 @@
 import { of } from 'rxjs';
 
-import { dateTime } from '@grafana/data';
-import { setBackendSrv } from '@grafana/runtime';
+import {
+  DataSourceInstanceSettings,
+  DataSourcePluginMeta,
+  PluginMetaInfo,
+  PluginType,
+  VariableHide,
+} from '@grafana/data';
+import { getBackendSrv, setBackendSrv } from '@grafana/runtime';
+import { getTimeSrv } from 'app/features/dashboard/services/TimeSrv';
 import { TemplateSrv } from 'app/features/templating/template_srv';
 import { initialCustomVariableModelState } from 'app/features/variables/custom/reducer';
 import { CustomVariableModel } from 'app/features/variables/types';
 
-import { TemplateSrvMock } from '../../../../features/templating/template_srv.mock';
 import { CloudWatchDatasource } from '../datasource';
+import { CloudWatchJsonData } from '../types';
+
+export function setupMockedTemplateService(variables: CustomVariableModel[]) {
+  const templateService = new TemplateSrv();
+  templateService.init(variables);
+  templateService.getVariables = jest.fn().mockReturnValue(variables);
+  return templateService;
+}
+
+const info: PluginMetaInfo = {
+  author: {
+    name: '',
+    url: undefined,
```

Comment:
if something is undefined, we could also just not define the property right? I think I tend to use null for things like this